core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -188,55 +202,97 @@ struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
           ? iteration_pattern::inner_direction
           : default_inner_direction<typename traits::execution_space>::value);
 
-  // Ugly ugly workaround intel 14 not handling scoped enum correctly
   static constexpr int Right = static_cast<int>(Iterate::Right);
   static constexpr int Left  = static_cast<int>(Iterate::Left);
 
-  KOKKOS_INLINE_FUNCTION const typename traits::execution_space& space() const {
-    return m_space;
-  }
-  template <typename LT, typename UT, typename TT = array_index_type>
-  MDRangePolicy(std::initializer_list<LT> const& lower,
-                std::initializer_list<UT> const& upper,
-                std::initializer_list<TT> const& tile = {})
-      : m_space() {
-    init(lower, upper, tile);
-  }
+  // </editor-fold> end Public member types and constexpr data members }}}2
+  //----------------------------------------------------------------------------
 
-  template <typename LT, typename UT, typename TT = array_index_type>
-  MDRangePolicy(const typename traits::execution_space& work_space,
-                std::initializer_list<LT> const& lower,
-                std::initializer_list<UT> const& upper,
-                std::initializer_list<TT> const& tile = {})
-      : m_space(work_space) {
-    init(lower, upper, tile);
-  }
+ private:
+
+  //----------------------------------------------------------------------------
+  // <editor-fold desc="Friends"> {{{2
+
+  template <class... OtherProperties>
+  friend struct MDRangePolicy;
+
+  // for now, make Impl::ParallelFor, etc., friends
+  // TODO design and implement a public interface for the things ParallelFor
+  //      and friends need
+
+  template <class, class, class>
+  friend class Kokkos::Impl::ParallelFor;
+
+  template <class, class, class, class>
+  friend class Kokkos::Impl::ParallelReduce;
+
+  template <class, class, class>
+  friend class Kokkos::Impl::ParallelScan;
+
+  template <class, class, class, class>
+  friend class Kokkos::Impl::ParallelScanWithTotal;
+
+  template <class, class, class, class, class>
+  friend struct Kokkos::Impl::HostIterateTile;
+
+#if defined(KOKKOS_ENABLE_CUDA)
+  template <int, class, class, class>
+  friend struct Kokkos::Impl::Refactor::DeviceIterateTile;
+  template <int, class, class, class, class, class>
+  friend struct Kokkos::Impl::Reduce::DeviceIterateTile;
+#endif
+
+  // </editor-fold> end Friends }}}2
```

Comment:
What do these comments mean? I have seen them a couple of time in PR.

Reply:
This makes it easier for IDEs to section off code. `<editor-fold>` is the Eclipse/CLion-style marker, `{{{` and `}}}` are the Vim-style markers.  I have no idea what the default marker is for emacs, otherwise I would include it.